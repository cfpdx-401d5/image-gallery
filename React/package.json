{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
