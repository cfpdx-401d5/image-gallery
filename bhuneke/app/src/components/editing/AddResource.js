import React, { Component } from 'react';
import fetcher from '../../helpers/fetcher';

const TITLE = 'title';
const DESCRIPTION = 'description';
const URL = 'url';

export default class AddResource extends Component {

    addRes(newResource) {
        return fetcher({
            path: '/resources',
            method: 'POST',
            body: {
                title: newResource.title,
                url: newResource.url,
                description: newResource.description,
            }
        });
    }

    render() {
        return (
            <div>
                <form onSubmit={(e) => {
                    e.preventDefault();
                    const newResource = {
                        //id: new Date(),
                        title: this.refs.title.value,
                        url: this.refs.url.value,
                        description: this.refs.description.value,                
                    };
                    this.addRes(newResource)
                        .then(res => {
                            this.props.onAdd(newResource);
                        })
                        .catch(err => {
                            console.log(err);
                        });

                }}>
                    <input ref={TITLE} required placeholder={TITLE}/>
                    <input ref={URL} required placeholder={URL}/>
                    <input ref={DESCRIPTION} required placeholder={DESCRIPTION}/>
                    <button>add</button>
                </form>
            </div>
        );
    }
}

