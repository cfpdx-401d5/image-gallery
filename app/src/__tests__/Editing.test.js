/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddResource from '../components/editing/AddResource';

describe('AddResource', () => {
    it('adds a resource', () => {
        let output = shallow(<AddResource onDelete={() => { } } />);
        expect(toJson(output)).toMatchSnapshot();
    });
});
