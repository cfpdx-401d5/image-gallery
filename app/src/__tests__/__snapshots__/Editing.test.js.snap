exports[`AddResource adds a resource 1`] = `
<div>
  <form
    onSubmit={[Function]}>
    <input
      placeholder="title"
      required={true} />
    <input
      placeholder="url"
      required={true} />
    <input
      placeholder="description"
      required={true} />
    <button>
      add
    </button>
  </form>
</div>
`;
